src/Agent.Worker/AgentPluginManager.cs

```diff
@@ -20,96 +18,46 @@ namespace Microsoft.VisualStudio.Services.Agent.Worker
-                    Type type = Type.GetType(pluginTypeName, throwOnError: true);
-                    taskPlugin = Activator.CreateInstance(type) as IAgentTaskPlugin;
-                }
-                finally
-                {
-                    AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
-                }
-
-                ArgUtil.NotNull(taskPlugin, nameof(taskPlugin));
-                ArgUtil.NotNull(taskPlugin.Id, nameof(taskPlugin.Id));
-                ArgUtil.NotNullOrEmpty(taskPlugin.Stage, nameof(taskPlugin.Stage));
-                if (!_supportedTasks.ContainsKey(taskPlugin.Id))
-                {
-                    _supportedTasks[taskPlugin.Id] = new List<string>();
-                }
-
-                Trace.Info($"Loaded task plugin id '{taskPlugin.Id}' ({taskPlugin.Stage}).");
-                _supportedTasks[taskPlugin.Id].Add(pluginTypeName);
-            }
-
-            // Load command plugin
-            foreach (var pluginTypeName in _commandPlugins)
-            {
-                IAgentCommandPlugin commandPlugin = null;
-                AssemblyLoadContext.Default.Resolving += ResolveAssembly;
-                try
-                {
-                    Trace.Info($"Load command plugin from '{pluginTypeName}'.");
-                    Type type = Type.GetType(pluginTypeName, throwOnError: true);
-                    commandPlugin = Activator.CreateInstance(type) as IAgentCommandPlugin;
+                    foreach (var type in assembly.GetTypes())
+                    {
+                        if (typeof(IAgentTaskPlugin).IsAssignableFrom(type) && !type.IsInterface && !type.IsAbstract)
+                        {
+                            IAgentTaskPlugin taskPlugin = Activator.CreateInstance(type) as IAgentTaskPlugin;
+                            ArgUtil.NotNull(taskPlugin, nameof(taskPlugin));
+                            ArgUtil.NotNull(taskPlugin.Id, nameof(taskPlugin.Id));
+                            ArgUtil.NotNullOrEmpty(taskPlugin.Stage, nameof(taskPlugin.Stage));
+                            if (!_supportedTasks.ContainsKey(taskPlugin.Id))
+                            {
+                                _supportedTasks[taskPlugin.Id] = new List<string>();
+                            }
+
+                            Trace.Info($"Loaded task plugin id '{taskPlugin.Id}' ({taskPlugin.Stage}).");
+                            var pluginTypeName = $"{type.FullName}, {assembly.FullName.Split(',')[0]}";
+                            _taskPlugins.Add(pluginTypeName);
+                            _supportedTasks[taskPlugin.Id].Add(pluginTypeName);
+                        }
+                    }
                 }
-                finally
+                catch (ReflectionTypeLoadException)
                 {
-                    AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
+                    // ignore exceptions for assemblies we cannnot load
```

Comment:
Why isn't this exceptional?

Reply:
there were some system assemblies that would get returned from ```AppDomain.CurrentDomain.GetAssemblies()``` but threw that exception when .GetTypes() was called on it. I can look more into it rather than just ignore that exception.

Reply:
Cool - I'm fine not blocking on it for this PR, but would be interested in understanding it!

Reply:
I found this explanation  https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.gettypes?view=netframework-4.8

Probably worth not ignoring that exception in case something was misconfigured in our code. Refactored a bit to not need the ignore.